ui/permissions.py

```diff
@@ -68,4 +204,7 @@ def has_permission(self, request, view):
             uuid.UUID(collection_key)
         except ValueError:
             raise ValidationError('wrong UUID format for {}'.format(collection_key))
-        return Collection.objects.filter(owner=request.user, key=collection_key).exists()
+        collection = Collection.objects.filter(key=collection_key)
+        if collection:
+            return has_admin_permission(collection.first(), request)
+        return False
```

Comment:
The check for `collection is not None` also prevents `has_admin_permission` from being run.

```python
import time
def expensive_function():
    time.sleep(5)
    return True
collections = None
# expensive_function doesn't run b/c collections is None
collections is not None and expensive_function()
```